adds zacks hack scripts

cheesesashimi opened the pull request:
**- What I did**
These are some small hack scripts I wrote to make it quicker to work on the MCD between build / run / test cycles.

**- How to verify it**

1. Provision an OpenShift cluster.
2. Store your kubeconfig in a file.
3. `export KUBECONFIG=/path/to/kubeconfig`.
4. Run `$ ./hack/prep-cluster-for-mcd-push.sh`.
5. Make a change to the MCD.
6. Run `$ ./hack/push-to-mcd-pods.sh`.

**- Description for the changelog**
MCO developer tool enhancements

cgwalters reviewed (commented):
Nice!  I only skimmed this, but I am all in favor of anything that improves iteration time!

cheesesashimi commented:
Trying to figure out an unrelated CI failure:

/test all

mkenigs commented on hack/push-to-mcd-pods.sh:
```diff
@@ -0,0 +1,120 @@
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+# shellcheck disable=SC1091
+source "$SCRIPT_DIR/hacklib"
```
```suggestion
source "$SCRIPT_DIR/lib"
```

mkenigs commented on hack/prep-cluster-for-mcd-push.sh:
```diff
@@ -0,0 +1,65 @@
+      containers:
+      - args:
+          - -c
+          - "sha256sum $ROOTFS_MCD_PATH && cp $ROOTFS_MCD_PATH /usr/local/bin/machine-config-daemon && /usr/local/bin/machine-config-daemon start -v 4"
```
I'm curious, why have you taken the approach of copying the binary rather than doing something more similar to `./hack/cluster-push.sh` but only pushing the new image for the MCD?

Related wondering: why do we lump all of our binaries into one image?

cheesesashimi replied:
I did it this way because it's a bit faster to build a single binary and push the binary instead of that approach. That's not to say that `./hack/cluster-push.sh` isn't valuable 😄 .

cheesesashimi replied:
I just remembered why I wrote these. I was trying to have a scripted way of doing the following: https://github.com/openshift/machine-config-operator/blob/master/docs/HACKING.md#developing-the-mcd-without-building-images. I'll update the docs to point to the scripts and explain how they work, etc.

mkenigs replied:
Yep makes sense! I was kinda thinking the increase in speed had to do with not changing operator/controller/server, but I guess building and pushing a container is slow

jkyros commented on hack/get-mcd-nodes.py:
```diff
@@ -0,0 +1,90 @@
+
+# This is a simple script that grabs all the cluster nodes and associates them
+# with a given MCD pod and outputs their roles. There is probably an easier way
+# to do this that I am too naïve to be aware of :).
```
Yeah. It's more apparent if you run it. I assume this was done because `oc get node` dosn't have the pods and `oc get pods` has the nodes but not the roles.
```
machine-config-daemon-9rf62	jkyros-fixed-buildcon-nqbcv-master-1      	master
machine-config-daemon-ggw9l	jkyros-fixed-buildcon-nqbcv-master-0      	master
machine-config-daemon-jjsql	jkyros-fixed-buildcon-nqbcv-master-2      	master
machine-config-daemon-nspjb	jkyros-fixed-buildcon-nqbcv-worker-c-z8zjg	worker
machine-config-daemon-sllxp	jkyros-fixed-buildcon-nqbcv-worker-d-hv4qb	worker
machine-config-daemon-wzvgk	jkyros-fixed-buildcon-nqbcv-worker-b-bkvtr	worker
```

cheesesashimi replied:
Yeah, I'll update that comment and include this sample output in said comment.

jkyros reviewed (approved):
I did try this and it does work. 

Quite a bit faster than waiting for the whole `hack/cluster_push.sh` dance to finish. This is pretty cool, thanks Zack. 

Might be worth dropping a line or two into https://github.com/openshift/machine-config-operator/blob/master/docs/HACKING.md when this goes in, but I wouldn't hold it for that.

kikisdeliveryservice commented:
@cheesesashimi can you squash the commits before merging?

cheesesashimi commented on hack/push-to-mcd-pods.sh:
```diff
@@ -0,0 +1,120 @@
+  # We don't have the file, so lets copy it
+  if [ "$has_file_retval" -ne 0 ]; then
+    echo "Binary not found on $pod"
+    copy_binary "$pod" "$bin_path" "$ROOTFS_MCD_PATH"
```
It doesn't. That was an oversight.

cheesesashimi commented:
Thanks for all the feedback! I did the following:
- Squashed my commits.
- Cleaned up unused variables.
- Updated the `docs/HACKING.md` doc with more information about the scripts and their intended use-cases.

kikisdeliveryservice commented:
will leave final review to John but

/approve

kikisdeliveryservice commented:
/assign @jkyros

jkyros commented:
Thanks for adding doc, looks good, and thanks again for this, it has already saved me some time :smile: 

/lgtm
